### PR TITLE
feat: adding Tasks API

### LINKED ISSUE #170: Using tasks
I see in the domain package the autogenerated `domain.TaskCreateRequest` but there is no corresponding handler in `api` package. So there is no way currently to programmatically make a task, control it, from this library?

### PR DESCRIPTION
**Pull request by vlastahajek**
Closes #170 

## Proposed Changes
Adding TasksAPI for managing tasks and associated logs and runs.

## Checklist

- [X] CHANGELOG.md updated
- [X] Rebased/mergeable
- [X] A test has been added if appropriate
- [X] Tests pass
- [X] Commit messages are in [semantic format](https://seesparkbox.com/foundry/semantic_commit_messages)

### CODE REVIEW

File: api/paging_test.go [hunk 1]
```diff
@@ -25,10 +25,10 @@ func TestPaging(t *testing.T) {
 	assert.Equal(t, domain.After("1111"), paging.after)
 
 	paging = &Paging{}
-	PagingWithLimit(0)(paging)
+	PagingWithLimit(1)(paging)
```
Comment (sranka) [inline api/paging_test.go:28]: PagingWithLimit(0)(paging) => assert.Equal(t, domain.Limit(0), paging.limit)

File: api/paging.go [hunk 2]
```diff
@@ -26,21 +26,16 @@ type Paging struct {
...
-// defaultPagingOptions returns default paging options: offset 0, limit 20, default sorting, ascending
+// defaultPagingOptions returns default paging options: offset 0, limit 0 (not applied), default sorting, ascending
 func defaultPaging() *Paging {
-	return &Paging{limit: 20, offset: 0, sortBy: "", descending: false, after: ""}
+	return &Paging{limit: 0, offset: 0, sortBy: "", descending: false, after: ""}
 }
 
 // PagingWithLimit sets limit option - maximum number of items returned.
-// Default 20, minimum 1 and maximum 100.
+// For buckets and organization the  default 20, minimum 1 and maximum 100,
```
Comment (sranka) [inline api/paging.go:35]: Is the default controlled by InfluxDB server. Wouldn't be better to express the defaults in buckets/orgs APIs.

**Review by sranka — DISMISSED** [review]
looks good to me

File: api/paging.go (hunk 2 above)
Comment (vlastahajek) [reply api/paging.go:35]: Ok. Orgs and Bucket API already have this comment

File: api/tasks.go [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+	// RemoveOwner removes an owner from a task.
+	RemoveOwner(ctx context.Context, task *domain.Task, user *domain.User) error
+	// RemoveOwner removes a member with id memberID from a task with taskID.
+	RemoveOwnerWithID(ctx context.Context, taskID, memberID string) error
+	// FindRuns retrieves a task runs according the filter. More fields can be applied. Filter can be nil.
+	FindRuns(ctx context.Context, task *domain.Task, filter *RunFilter) ([]domain.Run, error)
+	// FindRunsWithID retrieves runs of a task with taskID according the filter. More fields can be applied. Filter can be nil.
+	FindRunsWithID(ctx context.Context, taskID string, filter *RunFilter) ([]domain.Run, error)
+	// FindRunByID returns specific task run by taskID and runID
+	FindRunByID(ctx context.Context, taskID, runID string) (*domain.Run, error)
```
Comment (sranka) [inline api/tasks.go:101]: Can be easily confused with `FindRunsWithID` ... suggestion: rename to `GetRunByID`

**Review by sranka — APPROVED** [review]

File: api/tasks.go (hunk 3 above)
Comment (vlastahajek) [reply api/tasks.go]: Ok. For retrieving a single specific entity instance there are Get* and Get*ByID functions.

**Pull request merged**